core/tests/behavior/async_write.rs

```diff
@@ -634,12 +635,36 @@ pub async fn test_write_with_if_none_match(op: Operator) -> Result<()> {
 
     let (path, content, _) = TEST_FIXTURE.new_file(op.clone());
 
+    op.write(&path, content.clone())
+        .await
+        .expect("write must succeed");
+
+    let meta = op.stat(&path).await?;
+
+    let res = op
+        .write_with(&path, content.clone())
+        .if_none_match(meta.etag().expect("etag must exist"))
+        .await;
+    assert!(res.is_err());
+    assert_eq!(res.unwrap_err().kind(), ErrorKind::ConditionNotMatch);
+
+    Ok(())
+}
+
+/// Write an file with if_not_exists will get a ConditionNotMatch error if file exists.
+pub async fn test_write_with_if_not_exists(op: Operator) -> Result<()> {
+    if !op.info().full_capability().write_with_if_not_exists {
+        return Ok(());
+    }
+
+    let (path, content, _) = TEST_FIXTURE.new_file(op.clone());
+
     op.write(&path, content.clone())
         .await
         .expect("write must succeed");
     let res = op
         .write_with(&path, content.clone())
-        .if_none_match("*")
+        .if_not_exists(true)
```

Comment:
We can modify the test to write with `if_not_exists(true)` twice. The first will succeed, and the second will fail.